### DISCUSSION TITLE
What Does the "GR" in GRDB Stand For?

### DISCUSSION THREAD

**Discussion by codeface-io** [groue/GRDB.swift#1286] [Q&A] [answered]
Please make it easier for me and other people who discover GRDB to immediately "get it" i.e. click with it 😉

**Comment by groue** [comment, marked as answer]
Hello @flowtoolz!

I'm afraid those are... my initials, for Gwendal Roué!

For the archeologists who stumble on this page, I can give more details:

My first public open source library was a [Mustache](https://mustache.github.io) templating engine, written in pre-ARC (!) Objective-C, so I needed a prefix, and I lazily picked GR: **GR**Mustache (like **NS**String or **UI**View).

Then I wanted to start something with SQLite and Swift, for my coworkers and I. At that time[^1] the very popular SQLite wrapper was the Objective-C [FMDB](https://github.com/ccgus/fmdb). I wanted to help the transition from FMDB to GRDB, to make the life of FMDB users easy. So I grabbed many names from this library. For example, the `DatabaseQueue` and `Database` types are direct ports from FMDB. Even today, seven years later, you can ignore the fancy SQL builder, the record protocols, and just enjoy raw SQL with FMDB-like apis.

GRDB is a tribute to FMDB ❤️ 

[^1]: FMDB is *still* popular. For example, the excellent RSS reader [NetNewsWire](https://github.com/Ranchero-Software/NetNewsWire) [uses FMDB today](https://github.com/Ranchero-Software/NetNewsWire/blob/main/Technotes/Dependencies.markdown).

**Reply by codeface-io** [reply]
Thanks for the quick response! 🙏🏻

Maybe the name should be addressed up front, since it's cryptic to newcomers and not just because it's interesting to archeologists 😁

I know this thought comes about 5400 GitHub stars too late, but let me share it: I think a more catchy, expressive and inclusive/universal name would boost conversion/adoption even more. I didn't immediately realize GRDB is about databases. It looks like an acronym, similar to GDPR, and the "DB" part could stand out more. I believe people would fall in love with GRDB easier if it was called SwiftDB or something like that. But maybe that's just my personal aesthetic sensibility.

**Reply by groue** [reply]
Yes, I have thought about renaming the project on a few occasions, and make it more catchy.

Now, as you say, one does not throw away 5K stars and a reputation built over years in a whim. Also there exists tweets, blog posts, StackOverflow answers, Reddit comments, and generally online content and urls that I can not change, are indexed by Google, and are useful for library users.

For better of for worse, the library started as a personal project, with an old-fashioned name, and semi-private cryptic references to almost forgotten software. It is still, today, tied to my name, and did not turn into a community project that could be named on its own.
